### ISSUE TITLE
`Either<i32, f64>` incorrectly treats f64 as i32 in JavaScript binding

### ISSUE THREAD

**Issue by Daniel-Boll** [napi-rs/napi-rs#2325]
When using the `Either` enum from the `napi` crate to handle input values that could be either an `i32` or `f64`, the current implementation doesn't behave as expected for floating-point numbers. Specifically, the code below treats `f64` values as if they were `i32`, resulting in the truncation of the decimal part.

```rust
#[napi]
pub fn test(input: Either<i32, f64>) -> String {
  match input {
    Either::A(a) => format!("as i32: {}", a),
    Either::B(b) => format!("as f64: {}", b),
  }
}
```

When calling this from JavaScript:

```javascript
import { test } from '.'

console.log(test(1))     // as i32: 1
console.log(test(2.21))  // as i32: 2 (Expected: as f64: 2.21)
```

**Expected Behavior:**
- `test(1)` should return `"as i32: 1"`, which works correctly.
- `test(2.21)` should return `"as f64: 2.21"`, but it incorrectly returns `"as i32: 2"` because the input is being interpreted as `i32` instead of `f64`.

**Actual Behavior:**
- The function interprets both integer and floating-point inputs as `i32` when passed from JavaScript, even when a float value is supplied. This causes unintended behavior, particularly the truncation of `f64` values to `i32`.

**Comment by CPunisher** [comment]
Currently `Either<A, B>` first tries to convert js value to `A`. If it fails, then it tries to convert js value to `B`.
The conversion is performend by https://nodejs.org/api/n-api.html#napi_get_value_int32, which says:
> If the number exceeds the range of the 32 bit integer, then the result is truncated to the equivalent of the bottom 32 bits. This can result in a large positive number becoming a negative number if the value is > 231 - 1.

There is also a official issue about this problem: https://github.com/nodejs/node-addon-api/issues/578.

So I think it could be hard to fix this problem. Maybe you can directly use `f32` and manually check it?

**Comment by Daniel-Boll** [comment]
Yeah, I figured this could be a tough issue considering how NodeJS handles numbers internally. It seems like this is outside the scope of napi-rs, so I’ll need to explore some alternatives on my end. Thanks for the heads-up and suggestion on a potential solution. Feel free to close this issue if you’d like!
